### PR TITLE
Added multiple directories to search with qmk doctor command to resolve #9906

### LINKED ISSUE #9906: [Bug] QMK doctor only checks /etc/udev/rules.d for udev rules.
The doctor command in qmk python library fails to find the correct rules in `/usr/lib/udev/rules.d` directory.

## Describe the Bug

The doctor command in QMK python library only checks a hard coded `/etc/udev/rules.d` for udev rules. Which works fine if you put your rules there. But these rules can also be located in `/usr/lib/udev/rules.d`; and actually is the preferred location for packages to drop their udev rules. I recently packaged the [python client in archlinux](https://aur.archlinux.org/packages/python-qmk); and also provided neccessary udev rules along with it. After switching to these rules, I noticed I was getting warnings from `qmk doctor` when I had the correct udev rules in place.

## System Information

 - Keyboard: N/A
 - Operating system: Archlinux
 - AVR GCC version: 10.1.0
 - ARM GCC version: 10.1.0
 - QMK Firmware version: 0.9.46
 - Any keyboard related software installed? 
   - [ ] AutoHotKey
   - [ ] Karabiner
   - [ ] Other:

## Additional Context

I also implemented a quick fix; and will issue a pull request.

### PR DESCRIPTION
**Pull request by silverbluep**
Added `/usr/lib/udev/rules.d` as a directory to search for udev rules in the `qmk doctor` command

## Description

Added `/usr/lib/udev/rules.d` as a directory to search for udev rules in the `qmk doctor` command; and made the search go through udev rules accross all files. Currently; the search is hardcoded to only go through the `/etc/udev/rules.d` directory; and misses the appropriate rules if they are placed in the `/usr/lib/udev/rules.d/` directory.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes [9906](https://github.com/qmk/qmk_firmware/issues/9906)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Foxboron** [comment]
I'm in the process of adding `qmk_cli` to the Arch Linux repositories. This patch would be lovely to have for that.

**Comment by bbaserdem** [comment]
Thanks; @foxboron; did not know you used qmk. I added the files in my pkgbuild [here](https://aur.archlinux.org/packages/python-qmk/) if you wanna check it out.

File: lib/python/qmk/cli/doctor.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def check_udev_rules():
     """Make sure the udev rules look good.
     """
     ok = True
-    udev_dir = Path("/etc/udev/rules.d/")
+    udev_dir_list = (Path("/etc/udev/rules.d/"), Path("/lib/udev/rules.d"))
```
Comment (skullydazed) [inline lib/python/qmk/cli/doctor.py:150]:
```suggestion
    udev_dirs = (Path("/etc/udev/rules.d/"), Path("/lib/udev/rules.d"))
```

File: lib/python/qmk/cli/doctor.py [hunk 2]
```diff
@@ -170,30 +170,32 @@ def check_udev_rules():
...
-                    cli.log.warn("{bg_yellow}Detected ModemManager without the necessary udev rules. Please either disable it or set the appropriate udev rules if you are using a Pro Micro.")
-            if not rules.issubset(current_rules):
-                deprecated_rule = deprecated_rules.get(bootloader)
-                if deprecated_rule and deprecated_rule.issubset(current_rules):
-                    cli.log.warn("{bg_yellow}Found old, deprecated udev rules for '%s' boards. The new rules on https://docs.qmk.fm/#/faq_build?id=linux-udev-rules offer better security with the same functionality.", bootloader)
-                else:
-                    cli.log.warn("{bg_yellow}Missing udev rules for '%s' boards. You'll need to use `sudo` in order to flash them.", bootloader)
+    # Get configuration options of all config files
+    current_rules = set()
+    for udev_dir in udev_dir_list:
```
Comment (skullydazed) [inline lib/python/qmk/cli/doctor.py:175]:
```suggestion
    for udev_dir in udev_dirs:
```

File: lib/python/qmk/cli/doctor.py [hunk 3]
```diff
@@ -170,30 +170,32 @@ def check_udev_rules():
...
+        if udev_dir.exists():
+            udev_rules = [rule_file for rule_file in udev_dir.glob('*.rules')]
```
Comment (skullydazed) [inline lib/python/qmk/cli/doctor.py:177]:
```suggestion
            udev_rule_files = sorted(udev_dir.glob('*.rules'))
```

File: lib/python/qmk/cli/doctor.py [hunk 4]
```diff
@@ -170,30 +170,32 @@ def check_udev_rules():
...
+
+            # Collect all rules from this config file
+            for rule_file in udev_rules:
```
Comment (skullydazed) [inline lib/python/qmk/cli/doctor.py:180]:
```suggestion
            for rule_file in udev_rule_files:
```

File: lib/python/qmk/cli/doctor.py [hunk 5]
```diff
@@ -170,30 +170,32 @@ def check_udev_rules():
...
+                for line in rule_file.read_text().split('\n'):
+                    line = line.strip()
+                    if not line.startswith("#") and len(line):
```
Comment (skullydazed) [inline lib/python/qmk/cli/doctor.py:183]:
```suggestion
                    if line and not line.startswith("#"):
```

**Review by skullydazed — COMMENTED** [review]
I apologize for the delay getting to this review, 2020 has given me many surprises. 

There are some style things I noticed and unfortunately you'll have to rebase to fix the merge conflict, but once all of that is address this looks ready to go to me.

**Comment by silverbluep** [comment]
I did a rebase and implemented the suggested changes; however my fix seems to have regressed and even with the correct udev rules (in both directories) i'm getting warnings about using sudo. I will try to fix it; but I can't figure out why since the code block in question was not really changed throught the master branch commits.

**Comment by Erovia** [comment]
Hey @silverbluep, I'm so sorry for not checking this out sooner. Crazy year...
Anyway, there have been a number of changes in relations to the udev rules.
Both the rules changed a bit and now they are under `util/udev/50-qmk.rules` and not the documentation.

In any case, the `/usr/lib/udev/rules.d/` directory was left out intentionally, as it's used by the system and controlled by the distributions' packagers.
I don't know of any distribution that would ship all the rules we need.
So to speed things up, only the user-controlled `/etc/udev/rules.d` is scanned, which generally has a lot less rules.

**Comment by silverbluep** [comment]
> Hey @silverbluep, I'm so sorry for not checking this out sooner. Crazy year...
> Anyway, there have been a number of changes in relations to the udev rules.
> Both the rules changed a bit and now they are under `util/udev/50-qmk.rules` and not the documentation.
> 
> In any case, the `/usr/lib/udev/rules.d/` directory was left out intentionally, as it's used by the system and controlled by the distributions' packagers.
> I don't know of any distribution that would ship all the rules we need.
> So to speed things up, only the user-controlled `/etc/udev/rules.d` is scanned, which generally has a lot less rules.

Sorry for not seeing your reply sooner. Archlinux has the qmk package that ships the needed rules; and puts them to `/usr/lib/udev/rules.d/` since it's installed as a system package. If possible could this feature be readded? Previously, in my working fork, the scanning did not really impact runtime at all. (After rebasing, since it was a few months between me submitting this request and getting comments on it, my fork was failing silently and I could not figure out why.)

**Comment by fauxpark** [comment]
Needs a rebase

**Pull request closed without merging**